sdk/go/auto/cmd.go

```diff
@@ -281,6 +283,15 @@ func (p pulumiCommand) Run(ctx context.Context,
 	cmd.Stdout = io.MultiWriter(additionalOutput...)
 	cmd.Stderr = io.MultiWriter(additionalErrorOutput...)
 	cmd.Stdin = stdin
+	setSysprocAttrNewProcessGroup(cmd)
+	cmd.Cancel = func() error {
+		err := interruptProcess(cmd.Process)
+		if err != nil {
+			_ = cmd.Process.Kill()
```

Comment:
Codecov complains about missing coverage % here. I am happy to merge this as is, but if we really wanted to, we could have a test case that listens to `os.Interrupt` and does nothing in the handler. That should force us into the error case here.

Reply:
I don't think it's as easy as that for two reasons:
- The interrupt is set to the pulumi binary, not the program pulumi is running. Pulumi will abort the program even if the program ignores the interrupt, from my testing.
- Even if we could ignore the interrupt, we would not end up in the error case here.  This error only happens if the signal fails to be sent, which would be an OS level problem, which I think is too hard to reproduce in the test.

Reply:
Oh you're right about both points. Definitely too hard to reproduce this in a meaningful way.